static/js/reactions.js

```diff
@@ -501,32 +521,115 @@ export function set_clean_reactions(message) {
 
         message.clean_reactions.set(
             local_id,
-            make_clean_reaction({local_id, user_ids, ...reaction}),
+            make_clean_reaction({message, local_id, user_ids, ...reaction}),
         );
     }
 }
 
-function make_clean_reaction({local_id, user_ids, emoji_name, emoji_code, reaction_type}) {
+function make_clean_reaction({message, local_id, user_ids, emoji_name, emoji_code, reaction_type}) {
     const r = {
         local_id,
         user_ids,
         ...emoji.get_emoji_details_for_rendering({emoji_name, emoji_code, reaction_type}),
     };
 
-    update_user_fields(r);
+    update_user_fields(r, message);
+    r.vote_text = get_vote_text_in_clean_reactions(r.user_ids, message);
 
     r.emoji_alt_code = user_settings.emojiset === "text";
     r.is_realm_emoji = r.reaction_type === "realm_emoji" || r.reaction_type === "zulip_extra_emoji";
 
     return r;
 }
 
-export function update_user_fields(r) {
+export function update_user_fields(r, message) {
     r.count = r.user_ids.length;
+    r.vote_text = get_vote_text(r.user_ids, message);
     r.label = generate_title(r.emoji_name, r.user_ids);
     if (r.user_ids.includes(page_params.user_id)) {
         r.class = "message_reaction reacted";
     } else {
         r.class = "message_reaction";
     }
 }
+
+// to set vote text when message is being cleaned.
+// this is an alternative to the commented out 'if'
+// block in get_total_votes_on_message() for now.
+function get_vote_text_in_clean_reactions(user_list, message) {
+    const helpers = get_helpers;
+    if (!helpers().should_display_emoji_reaction_users()) {
+        return `${user_list.length}`;
+    }
+    const usernames = people.get_display_full_names(user_list);
+    const current_user_has_reacted = user_list.includes(page_params.user_id);
+
+    if (current_user_has_reacted) {
+        const current_user_index = user_list.indexOf(page_params.user_id);
+        usernames[current_user_index] = $t({
+            defaultMessage: "You",
+        });
+    }
+
+    if (message.reactions.length <= 3) {
+        const comma_separated_usernames = usernames.join(", ");
+        return comma_separated_usernames;
+    }
+
+    // returning length as a string instead of integer for consistency
+    return `${usernames.length}`;
+}
+
+export function get_vote_text(user_list, message) {
+    const helpers = get_helpers;
+    if (!helpers().should_display_emoji_reaction_users()) {
```

Comment:
Could we spell this as 
`if (!helpers().should_display_emoji_reaction_users() || get_total_votes_on_message(message) > 3)`?  It seems like all the code below here is for the case where we're definitely displaying the names, and that'd be more readable.